R/api.R

```diff
@@ -16,109 +16,379 @@
   get_content(hn_api_response)
 }
 
-#' Retrieves the item corresponding to the specified id
+
+#' @title Print for "hackerlist" type objects
+#'
+#' @param x "hackerlist" type list
+#'
+#' @param ... further arguments passed to or from other methods
+#'
+#' @export
+#'
+#' @examples
+#'
+#' hacker_news <- get_top_stories(max_items = 5)
+#' print(hacker_news)
+
+print.hackerlist <- function(x, ...) {
+  cat('Title:', x$title, '\n')
+  cat('ID:', x$id, '\n')
+  cat('Author:', x$by, '\n')
+  cat('Descendants:', x$descendants, '\n')
+  cat('Kids: ')
+  if(length(x$kids) > 5)
+    cat(utils::head(x$kids),'... \n')
+  else
+    cat(x$kids, '\n')
+  cat('Score:', x$score,'\n')
+  cat('Date:', as.character(x$time),'\n')
+  cat('Type:', x$type,'\n')
+  cat('URL:', x$url,'\n')
+}
+
+#' @title Print for "hackeruser" type objects
+#'
+#' @param x "hackerlist" type list
+#'
+#' @param ... further arguments passed to or from other methods
 #'
 #' @export
 #'
+#' @examples
+#'
+#' hacker_user <- get_user_by_username('steveddaniels')
+#' print(hacker_user)
+
+print.hackeruser <- function(x, ...) {
+  cat('ID:', x$id, '\n')
+  cat('Created:', as.character(x$created), '\n')
+  cat('Karma:', x$karma,'\n')
+  cat('Submitted: ')
+  if(length(x$submitted) > 5)
+    cat(utils::head(x$submitted),'... \n')
+  else
+    cat(x$submitted, '\n')
+}
+
+
+#' @title Get Hacker News by Id
+#'
+#' @description Retrieves the item corresponding to specified id using Hacker News API
+#'
 #' @param id id of the item that should be retrieved
 #'
+#' @export
+#'
 #' @return item correspoing to the specified id
 #'
+#' @examples
+#'
+#' h_id <- get_max_item_id()
+#' hacker_article <- get_item_by_id(h_id)
+#'
 get_item_by_id <- function(id) {
   request_url <- create_request_url(.base_url(), c("item", id))
-  .send_request(request_url)
+  item <- .send_request(request_url)
+  if(!is.null(item))
+  {
+    class(item) <- 'hackerlist'
+    item$time <- as.POSIXct(item$time, origin="1970-01-01")
+  }
+  item
 }
 
-#' Get user corresponding to the specified username
+
+#' @title Get Hacker News user
 #'
-#' @export
+#' @description Get user corresponding to the specified username
 #'
 #' @param username of the user that should be retrieved
 #'
+#' @export
+#'
 #' @return user corresponding to the specified username
 #'
+#' @examples
+#' user <- get_user_by_username("steveddaniels")
+#' print(user)
 get_user_by_username <- function(username) {
   request_url <- create_request_url(.base_url(), c("user", username))
-  .send_request(request_url)
+  user <- .send_request(request_url)
+  class(user) <- 'hackeruser'
```

Comment:
Same as with hackerlist. We need a separate constructor for this: http://adv-r.had.co.nz/S3.html